PresCrypt-Backend/PresCrypt.Application/Services/UserServices/UserServices.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using PresCrypt_Backend.PresCrypt.API.Dto;
+using PresCrypt_Backend.PresCrypt.Core.Models;
+
+namespace PresCrypt_Backend.PresCrypt.Application.Services.UserServices
+{
+    public class UserServices
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IPasswordHasher<User> _passwordHasher;
+
+        public UserServices(ApplicationDbContext context, IPasswordHasher<User> passwordHasher)
+        {
+            _context = context;
+            _passwordHasher = passwordHasher;
+        }
+
+        public async Task<string> RegisterUserAsync(UserRegDTO userRegDTO)
+        {
+            if (_context.User.Any(x => x.UserName.Equals(userRegDTO.UserName, StringComparison.OrdinalIgnoreCase)))
+            {
+                return "Email already exists.";
+            }
+
+            // Generate User ID (Uxxx)  
+            var lastUser = await _context.User.OrderByDescending(u => u.UserId).FirstOrDefaultAsync();
+            int newUserIdNum = lastUser != null && int.TryParse(lastUser.UserId.Substring(1), out int lastId) ? lastId + 1 : 1;
+            string newUserId = $"U{newUserIdNum:D3}";
+
+            var newUser = new User
+            {
+                UserId = newUserId,
+                UserName = userRegDTO.UserName,
+                PasswordHash = _passwordHasher.HashPassword(null, userRegDTO.Password),
+                Role = userRegDTO.Role,
+                Patient = new List<Patient>(),
```

Comment:
this is not the best practice. you should have user implemented for these sub classes.